docs/filtering-config.md

```diff
@@ -0,0 +1,54 @@
+# Filtering Drupal Configuration
+
+When exporting and importing configuration from and to a Drupal 8 site,
+Drush provides a mechanism called the Configuration Filter system which 
+allows configuration values to be altered during import and export, allowing 
+you to vary your configuration by environment.  The --skip-modules option
+in the config-import and config-export commands is implemented with a
+configuration filter.  For more complex uses, you will need to write some
+custom code.
+
+## Getting started
+
+The first thing that you will need to do is set up a Drush extension
+to hold your storage filter hook.  See the example
+[example sandwich commandfile](../examples/sandwich-drush.inc) for
+details; note, however, that it is not necessary for your commandfile
+to implement hook_drush_command(), or any other hook besides the storage
+filter hook.
+
+You will need a composer.json file as well, in order to define where
+your StorageFilter class is defined.  Make sure that Drush and your
+custom commandfile are required from the composer.json file of any
+Drupal site that you plan on using your filter with.
+
+## Implementing the Storage Filter Hook
```

Comment:
Give the existence of module overrides (https://www.drupal.org/node/1928898), I'm no longer sure if this hook system is needed. Perhaps the support burden is not worth it?